examples/minroot.rs

```diff
@@ -19,7 +19,58 @@ use std::time::Instant;
 use tracing_subscriber::{fmt, prelude::*, EnvFilter, Registry};
 use tracing_texray::TeXRayLayer;
 
-#[derive(Clone, Debug)]
+#[cfg(feature = "abomonate")]
+mod utils {
+  use super::*;
+  use std::{io::Write, mem::size_of};
+
+  pub const FILEPATH: &str = "/tmp/data";
+  /// Unspeakable horrors from a type safety PoV
+  unsafe fn any_as_u8_slice<T: Sized>(p: &T) -> &[u8] {
+    ::core::slice::from_raw_parts((p as *const T) as *const u8, ::core::mem::size_of::<T>())
+  }
+
+  /// this is dangerous
+  #[allow(non_snake_case)]
+  unsafe fn entomb_F<F: PrimeField, W: Write>(f: &F, bytes: &mut W) -> std::io::Result<()> {
+    println!("entomb: {}", size_of::<F>());
+    // this is  dangerous
+    bytes.write_all(any_as_u8_slice(&f))?;
+    Ok(())
+  }
+
+  /// this is dangerous
+  #[allow(non_snake_case)]
+  unsafe fn exhume_F<'a, F: PrimeField>(f: &mut F, bytes: &'a mut [u8]) -> Option<&'a mut [u8]> {
+    let (mine, rest) = bytes.split_at_mut(size_of::<F>());
+    let mine = (mine as *const [u8]) as *const F;
+    std::ptr::write(f, std::ptr::read(mine));
+    Some(rest)
+  }
```

Comment:
Can this function return `None`?

Reply:
No. The reason it is wrapped in option in this way is that it is meant to emulate the implementation of a trait (Abomonate) that requires this API. Nonetheless, I've added a commit that removes the wrapping.